operator-common/src/main/java/io/strimzi/operator/common/model/PemAuthIdentity.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common.model;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.strimzi.operator.common.Util;
+
+import java.io.ByteArrayInputStream;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Optional;
+
+/**
+ * Represents the identity used during TLS client authentication.
+ * This consists of an X509 end entity certificate, corresponding private key, and a chain of X509 CA certificates, all in PEM format.
+ */
+public class PemAuthIdentity {
+
+    private final byte[] pemPrivateKey;
+    private final byte[] pemCertificateChainBytes;
+    private final X509Certificate pemCertificateChain;
+    private String secretName;
+    private String secretNamespace;
+
+    /**
+     * @param secret Kubernetes Secret containing the Cluster Operator public and private key
+     * @param secretKey Key in the Kubernetes Secret that is associated with the requested identity
+     */
+    private PemAuthIdentity(Secret secret, String secretKey) {
+        Optional.ofNullable(secret)
+                .map(Secret::getMetadata)
+                .ifPresent(objectMeta -> {
+                    secretName = objectMeta.getName();
+                    secretNamespace = objectMeta.getNamespace();
+                });
+        pemPrivateKey = Util.decodeFromSecret(secret, String.format("%s.key", secretKey));
+        pemCertificateChainBytes = Util.decodeFromSecret(secret, String.format("%s.crt", secretKey));
+        pemCertificateChain = validateCertificateChain(secretKey);
+    }
+
+    /**
+     * Create a new instance of PemAuthIdentity that represents the identity of the
```

Comment:
I'm being very pedantic, but it's not relevant whether or not this returns a new instance:

```suggestion
     * Returns the PemAuthIdentity that represents the identity of the
```

Also If you say "create" or "return" that tells you one thing a piece of code code, but it doesn't preclude it doing something else. If you say "Returns" then it's slightly stronger, because it implies that that's _all_ it does.